lib/guess_checker.rb

```diff
@@ -0,0 +1,48 @@
+class GuessChecker
+  attr_reader :current_guess, :original_code, :check_code, :pins
+
+  def initialize(original_code)
+    @current_guess = ""
+    @original_code = original_code
+    @check_code = ""
+    @pins = {}
+  end
+
+  def check_red(guess, code)
+    red_count = 0
+    guess.length.times do |i|
+      if guess[i] == code[i]
+        red_count += 1
+        guess[i], code[i] = "-", "-"
+      end
+    end
+    red_count
+  end
+
+  def check_white(guess, code)
+    white_count = 0
+    guess.length.times do |i|
+      if guess[i].match(/[A-Z]/)
```

Comment:
Why are we doing a regex match here? `check_white` seems to be doing a little too much maybe?